Form/Extension/DependencyInjectionExtension.php

```diff
@@ -69,15 +64,40 @@ class DependencyInjectionExtension implements FormExtensionInterface
     private $guesserLoaded = false;
 
     /**
+     * @var array
+     */
+    private $mappingExtensionTypes;
+
+    /**
+     * @var array|bool
+     */
+    private $reverseMappingTypes;
+
+    /**
+     * DependencyInjectionExtension constructor.
+     *
      * @param ContainerInterface $container
      * @param array              $typeServiceIds
      * @param array              $typeExtensionServiceIds
      * @param array              $guesserServiceIds
      * @param array              $mappingTypes
      * @param array              $extensionTypes
      */
-    public function __construct(ContainerInterface $container, array $typeServiceIds, array $typeExtensionServiceIds, array $guesserServiceIds, array $mappingTypes = array(), array $extensionTypes = array())
-    {
+    public function __construct(
+        ContainerInterface $container,
+        array $typeServiceIds,
+        $typeExtensionServiceIds,
+        $guesserServiceIds,
+        array $mappingTypes = [],
+        array $extensionTypes = []
+    ) {
+        if (is_object($typeExtensionServiceIds) && $typeExtensionServiceIds instanceof \Iterator) {
+            $typeExtensionServiceIds = iterator_to_array($typeExtensionServiceIds);
+        }
+        if (is_object($guesserServiceIds) && $guesserServiceIds instanceof \Iterator) {
+            $guesserServiceIds = iterator_to_array($guesserServiceIds);
+        }
+
```

Comment:
You should throw a deprecation if the old way of calling this constructor is used.